internal/provider/data_source_fmc_device_ha_pairs_test.go

```diff
@@ -0,0 +1,155 @@
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "lan_failover_ipv6_addr", "false"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "lan_failover_interface_name", "GigabitEthernet0/0"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "lan_failover_interface_id", "<interface uuid>"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "lan_failover_interface_type", "PhysicalInterface"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_standby_ip", "1.1.1.2"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_active_ip", "1.1.1.1"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_name", "Stateful-INTERFACE"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_subnet_mask", "255.255.255.0"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_ipv6_addr", "false"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_interface_name", "GigabitEthernet0/0"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_interface_id", "<interface uuid>"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "stateful_failover_interface_type", "PhysicalInterface"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "action", "SWITCH"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.fmc_device_ha_pairs.test", "force_break", "false"))
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceFmcDeviceHAPairsConfig(),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+			{
+				Config: testAccNamedDataSourceFmcDeviceHAPairsConfig(),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+		},
+	})
+}
+
+// End of section. //template:end testAccDataSource
+
+// Section below is generated&owned by "gen/generator.go". //template:begin testPrerequisites
+// End of section. //template:end testPrerequisites
+
+// Section below is generated&owned by "gen/generator.go". //template:begin testAccDataSourceConfig
+
+func testAccDataSourceFmcDeviceHAPairsConfig() string {
+	config := `resource "fmc_device_ha_pairs" "test" {` + "\n"
+	config += `	name = "FTD_HA"` + "\n"
+	config += `	primary_device_id = "<FTD1_ID>"` + "\n"
```

Comment:
This test will fail, as the resource definition here cannot be met. 
Device ID should be taken from environmental variables and this field value overriden by test_value in yaml. Please refer to other Device related resources for samples.

Reply:
done